src/modules/deal/deals.service.ts

```diff
@@ -0,0 +1,181 @@
+import { Injectable } from '@nestjs/common';
+import { DEFAULTS } from '@ukef/constants/properties';
+import { AcbsService } from '@ukef/modules/acbs/acbs.service';
+import { AcbsAuthenticationService } from '@ukef/modules/acbs/acbs-authentication.service';
+
+import { CreateDealDto } from './dto/deals-request.dto';
+
+@Injectable()
+export class DealsService {
+  constructor(private readonly acbsAuthenticationService: AcbsAuthenticationService, private readonly acbsService: AcbsService) {}
+
+  async createDeal(createDealDto: CreateDealDto): Promise<any> {
+    const idToken = await this.acbsAuthenticationService.getIdToken();
+
+    const todayDate = new Date();
+    todayDate.setHours(0, 0, 0, 0); // Set to midnight
+    const todayDateAsIso = todayDate.toISOString();
+
+    const guaranteeCommencementDate = new Date(createDealDto.guaranteeCommencementDate);
+    guaranteeCommencementDate.setHours(0, 0, 0, 0);
+
+    const effectiveDate = guaranteeCommencementDate < todayDate ? guaranteeCommencementDate : todayDate;
```

Comment:
`effectiveDate` I believe will also be used in `\facility` endpoints, might be best to convert this to a helper function as well please.